manifests/rhsm_reconfigure_script.pp

```diff
@@ -0,0 +1,65 @@
+# @summary Compose RHSM reconfigure script via concats
+# @api private
+define foreman_proxy_content::rhsm_reconfigure_script(
+  Stdlib::Absolutepath $ca_cert = undef,
+  Stdlib::Absolutepath $server_ca_cert = undef,
+  Stdlib::Absolutepath $rhsm_ca_dir = undef,
+  String $default_ca_name = undef,
+  String $server_ca_name = undef,
```

Comment:
In Puppet it's not strictly needed to have a default. Since the default is invalid for the type, you can leave off `= undef` to indicate they're required parameters. That was done in Puppet < 4 before there were data types but is still entirely valid.